src/utils.ts

```diff
@@ -26,16 +26,29 @@ function loadTsFile(filename: string, cwd: string, tsConfig?: any): any {
   let { data, path } = joycon.loadSync([filename], cwd)
   if (path && data) {
     if (tsConfig) {
-      data = deepmerge(tsConfig, data)
+      data = deepmerge(data, tsConfig)
```

Comment:
Flipped the deepmerg args, from their docs `If an element at the same key is present for both x and y, the value from y will appear in the result.`. We want the data from the passed in `tsConfig` arg to not be overwritten preserving normal behaviour.

Reply:
But is this the normal behaviour of tsconfig itself? I think the current file's config will overwrite the config to be extended?

Reply:
What you said is correct. What I meant is previous behaviour would overwrite the original config.
In the main branch if the merge did what it is supposed to do then the base tsconfig params would be overwritten.

In the loadTsFile  the `tsConfig` arg is the arg being passed in, the `data` is the fetched, which on 2nd iteration of the function would be the tsConfig of the `extend` param. So in the master branch the merge would overwrite the incoming config from fn param with the newly read `data` from fs.

Reply:
Ah yes. Sorry my bad. Actually the incoming `tsConfig` is our user's one.

Reply:
The reason why the existing test worked in the first place for extending config is because the function had a bug where no extended tsconfig would be merged anyway.